packages/components/src/list/index.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * WordPress dependencies
+ */
+import { createElement, forwardRef } from '@wordpress/element';
+
+const List = ( { type = 'unordered', accessibilityLabel, ...props }, ref ) => {
+	const tagName = type === 'ordered' ? 'ol' : 'ul';
```

Comment:
If it's a "one or the other" decision, I think a boolean value would be preferable here, e.g. `isOrdered`. These sorts of magic strings could be difficult to work with.

Reply:
> A question I had coming into this is specifics around handling different types of list (ol, ul, maybe dl?), and the relationship between the parent element and the child element (li,dd,dt). Should we want to complete this abstraction with a corresponding ListItem element as well?

The answer is highly coupled with what you shared in the comment cited. My understanding was that we could support multiple types:
- unordered - `ul`
- ordered - `ol`
- description - `dl`
- unstyled - I suppose `ul`(?) but all styles removed

With that, it would be great to have `ListItem` component which would handle tag names dance internally (`li` for `ul` and `ol`, `dd` and `dt` for `dl`). In addition, it would make it possible to inherit those types for nested lists from the parents and so on. I'm sure there would be more benefits discovered.